gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/LayerConverter.java

```diff
@@ -33,10 +35,12 @@ static Layer toLayer(String id, LayerNestedObject obj) {
     if (obj.getDefaultStyle() != null) {
       layer.setDefaultStyle(StyleConverter.toStyle(obj.getDefaultStyle()));
     }
-    if (obj.getForms() != null) {
-      Maps.forEach(
-          obj.getForms(),
-          (formId, formDoc) -> layer.addForm(FormConverter.toForm(formId, formDoc)));
+    if (obj.getForms() != null && !obj.getForms().isEmpty()) {
```

Comment:
That's exactly right.. It would if we controlled how the field was populated, but since we don't know if the Firebase API will set a null or empty we need to check both.